packages/wonder-blocks-form/components/checkbox-core.js

```diff
@@ -0,0 +1,195 @@
+
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Color, {mix, fade} from "@khanacademy/wonder-blocks-color";
+import {View, addStyle} from "@khanacademy/wonder-blocks-core";
+import Icon from "@khanacademy/wonder-blocks-icon";
+
+import type {ClickableHandlers} from "@khanacademy/wonder-blocks-core";
+import type {IconAsset} from "@khanacademy/wonder-blocks-icon";
+
+import type {ChoiceCoreProps} from "../util/types.js";
+
+type Props = {|
+    ...ChoiceCoreProps,
+    ...ClickableHandlers,
+    hovered: boolean,
+    focused: boolean,
+    pressed: boolean,
+|};
+
+const {blue, red, white, offWhite, offBlack16, offBlack32, offBlack50} = Color;
+
+const StyledInput = addStyle("input");
+
+const checkboxCheck: IconAsset = {
+    small:
+        "M11.263 4.324a1 1 0 1 1 1.474 1.352l-5.5 6a1 1 0 0 1-1.505-.036l-2.5-3a1 1 0 1 1 1.536-1.28L6.536 9.48l4.727-5.157z",
+};
+
+export default class CheckboxCore extends React.Component<Props> {
+    render() {
+        const {
+            checked,
+            disabled,
+            error,
+            id,
+            groupName,
+            testId,
+            style,
+            hovered,
+            focused,
+            pressed,
+            ...handlers
+        } = this.props;
+
+        const stateStyles = _generateStyles(checked, error);
+
+        const defaultStyle = [
+            sharedStyles.inputReset,
+            sharedStyles.default,
+            stateStyles.default,
+            !disabled &&
+                (pressed
+                    ? stateStyles.active
+                    : (hovered || focused) && stateStyles.focus),
+            disabled && sharedStyles.disabled,
+        ];
+
+        const props = {
+            "data-test-id": testId,
+            ...handlers,
+        };
+
+        return (
+            <View style={[sharedStyles.wrapper, style]} {...props}>
+                <StyledInput
+                    type="checkbox"
+                    aria-checked={checked}
+                    checked={checked}
+                    disabled={disabled}
+                    id={id}
+                    name={groupName}
+                    // Need to specify because this is a controlled React
+                    // form component, but we handle the click via
+                    // ClickableBehavior already
+                    onChange={() => void 0}
+                    style={defaultStyle}
+                    tabIndex={-1}
+                />
+                {checked && (
+                    <Icon
+                        color={disabled ? offBlack32 : white}
+                        icon={checkboxCheck}
+                        size="small"
+                        style={sharedStyles.checkIcon}
+                    />
+                )}
+            </View>
+        );
+    }
+}
+
+const sharedStyles = StyleSheet.create({
+    wrapper: {
+        outline: "none",
+    },
+
+    // Reset the default styled input element
+    inputReset: {
```

Comment:
This seems like a good candidate to move into the `overrides` stylesheet in `add-style.js`.

Reply:
This style isn't in the normalize sheet. It's a reset on the browser-specific styles that are added to checkboxes and radio buttons. It may be the cause that we don't need them to apply for all `input` objects.